pylint/checkers/dunder_methods.py

```diff
@@ -167,6 +173,29 @@ def within_dunder_def(node: nodes.NodeNG) -> bool:
             parent = parent.parent
         return False
 
+    @only_required_for_messages("bad-dunder-name")
+    def visit_functiondef(self, node: nodes.FunctionDef) -> None:
+        """Check if known dunder method is misspelled or dunder name is not one
+        of the pre-defined names.
+        """
+        # ignore module-level functions
+        if not node.is_method():
+            return
+
+        if (
+            # Detect something that could be a dunder method
+            node.name.startswith("_")
+            or node.name.endswith("_")
```

Comment:
```suggestion
            node.name.startswith("_")
            and node.name.endswith("_")
```
A function starting with ``_`` is protected, a function starting with ``__`` is private those should not raise.